selfdrive/locationd/calibrationd.py

```diff
@@ -99,47 +100,45 @@ def handle_cam_odom(self, trans, rot, trans_std, rot_std):
     certain_if_calib = ((np.arctan2(trans_std[1], trans[0]) < MAX_VEL_ANGLE_STD) or
                         (self.valid_blocks < INPUTS_NEEDED))
     if straight_and_fast and certain_if_calib:
-      # intrinsics are not eon intrinsics, since this is calibrated frame
-      intrinsics = intrinsics_from_vp(self.vp)
-      new_vp = intrinsics.dot(view_frame_from_device_frame.dot(trans))
-      new_vp = new_vp[:2]/new_vp[2]
-      new_vp = sanity_clip(new_vp)
+      observed_rpy = np.array([0,
+                               -np.arctan2(trans[2], trans[0]),
+                               np.arctan2(trans[1], trans[0])])
+      new_rpy = euler_from_rot(rot_from_euler(self.rpy).dot(rot_from_euler(observed_rpy)))
+      new_rpy = sanity_clip(new_rpy)
 
-      self.vps[self.block_idx] = (self.idx*self.vps[self.block_idx] + (BLOCK_SIZE - self.idx) * new_vp) / float(BLOCK_SIZE)
+      self.rpys[self.block_idx] = (self.idx*self.rpys[self.block_idx] + (BLOCK_SIZE - self.idx) * new_rpy) / float(BLOCK_SIZE)
       self.idx = (self.idx + 1) % BLOCK_SIZE
       if self.idx == 0:
         self.block_idx += 1
         self.valid_blocks = max(self.block_idx, self.valid_blocks)
         self.block_idx = self.block_idx % INPUTS_WANTED
       if self.valid_blocks > 0:
-        self.vp = np.mean(self.vps[:self.valid_blocks], axis=0)
+        self.rpy = np.mean(self.rpys[:self.valid_blocks], axis=0)
       self.update_status()
 
       if self.param_put and ((self.idx == 0 and self.block_idx == 0) or self.just_calibrated):
-        calib = get_calib_from_vp(self.vp)
-        cal_params = {"calib_radians": list(calib),
+        cal_params = {"calib_radians": list(self.rpy),
```

Comment:
both of these values are in liveCalibration, why don't we just use that?

Reply:
Not quite sure what you mean? We still need to save calibration so we can load it when the process restarts.

Reply:
this json thing is just a subset of the struct defined in cereal. why don't we just save the last sent struct to a param instead of this new one?

Reply:
Oh then sure, can you do that?

Reply:
yeah, let’s merge this and I’ll do it in the polling PR

Reply:
yeah, let’s merge this and I’ll do it in the polling PR

Reply:
ok cool